src/streamlink/plugins/artetv.py

```diff
@@ -45,44 +44,18 @@ class ArteTV(Plugin):
     def can_handle_url(cls, url):
         return _url_re.match(url)
 
-    def _create_stream(self, stream, language):
-        stream_name = "{0}p".format(stream["height"])
-        stream_type = stream["mediaType"]
-        stream_url = stream["url"]
-        stream_language = stream["versionShortLibelle"]
-
-        if language == "de":
-            language = ["DE", "VOST-DE", "VA", "VOA", "Dt. Live", "OV", "OmU"]
-        elif language == "en":
-            language = ["ANG", "VOST-ANG"]
-        elif language == "es":
-            language = ["ESP", "VOST-ESP"]
-        elif language == "fr":
-            language = ["FR", "VOST-FR", "VF", "VOF", "Frz. Live", "VO", "ST mal"]
-        elif language == "pl":
-            language = ["POL", "VOST-POL"]
-
-        if stream_language in language:
-            if stream_type in ("hls", "mp4"):
-                if urlparse(stream_url).path.endswith("m3u8"):
+    def _create_stream(self, streams):
+        variant, variantname = min([(stream["versionProg"], stream["versionLibelle"]) for stream in streams], key=itemgetter(0))
+        log.debug("Using the '{0}' stream variant".format(variantname))
+        for stream in streams:
+            if stream["versionProg"] == variant:
+                if stream["mediaType"] == "hls":
                     try:
-                        streams = HLSStream.parse_variant_playlist(self.session, stream_url)
-
-                        for stream in streams.items():
-                            yield stream
-                    except IOError as err:
-                        self.logger.error("Failed to extract HLS streams: {0}", err)
-                else:
-                    yield stream_name, HTTPStream(self.session, stream_url)
-
-            elif stream_type == "f4m":
-                try:
-                    streams = HDSStream.parse_manifest(self.session, stream_url)
-
-                    for stream in streams.items():
-                        yield stream
-                except IOError as err:
-                    self.logger.error("Failed to extract HDS streams: {0}", err)
+                        streams = HLSStream.parse_variant_playlist(self.session, stream["url"])
+                        for s in streams.items():
+                            yield s
+                    except IOError:
+                        log.exception("Failed to extract HLS streams")
```

Comment:
```py
log.exception
```

this will add a **new line**, which looks weird.

Reply:
I changed it to a warning and included some more useful information.